browser/ui/sidebar/sidebar_controller.cc

```diff
@@ -134,8 +135,6 @@ void SidebarController::LoadAtTab(const GURL& url) {
 
 void SidebarController::OnShowSidebarOptionChanged(
     SidebarService::ShowSidebarOption option) {
-  // Clear active state whenever sidebar enabled state is changed.
-  ActivateItemAt(-1);
```

Comment:
My assumption is `ActivateItemAt(-1);` was deleted to make sidebar panel(+sidebar control view) open till user closes opened panel explicitely even if user changed the show option. @petemill Am I right?

But, sidebar control view doesn't get hidden after `never` option is set via context menu even if panel is not opened. I think this is reggression.

Reply:
I found the cause of this regression and seems fixed with below changes.
`ShouldShowSidebar()` returns true because context menu was not hidden.
```
diff --git a/browser/ui/views/sidebar/sidebar_container_view.cc b/browser/ui/views/sidebar/sidebar_container_view.cc
index 9e62f62627..2a03f8f3b8 100644
--- a/browser/ui/views/sidebar/sidebar_container_view.cc
+++ b/browser/ui/views/sidebar/sidebar_container_view.cc
@@ -125,11 +125,11 @@ void SidebarContainerView::SetSidebarShowOption(
   }
   // We don't set false explicitly just in case a panel view is still open.
   if (show_option == ShowSidebarOption::kShowNever) {
-    ShowSidebar(ShouldShowSidebar(), false);
+    ShowSidebar(IsPanelActive(), false);
     return;
   }
 
-  ShowSidebar(ShouldShowSidebar(), true);
+  ShowSidebar(IsPanelActive(), true);
 }
 
 void SidebarContainerView::UpdateSidebar() {
@@ -184,10 +184,7 @@ void SidebarContainerView::OnThemeChanged() {
   UpdateBackground();
 }
 
-bool SidebarContainerView::ShouldShowSidebar() const {
-  // Always show if panel should be visible. It is more reliable to check
-  // whether the active index is a panel item rather than checking if
-  // side_panel_ is visible.
+bool SidebarContainerView::IsPanelActive() const {
   auto active_index = sidebar_model_->active_index();
   bool panel_is_active = false;
   if (active_index != -1) {
@@ -197,8 +194,14 @@ bool SidebarContainerView::ShouldShowSidebar() const {
       panel_is_active = true;
     }
   }
+  return panel_is_active;
+}
 
-  return panel_is_active ||
+bool SidebarContainerView::ShouldShowSidebar() const {
+  // Always show if panel should be visible. It is more reliable to check
+  // whether the active index is a panel item rather than checking if
+  // side_panel_ is visible.
+  return IsPanelActive() ||
          sidebar_control_view_->IsItemReorderingInProgress() ||
          sidebar_control_view_->IsBubbleWidgetVisible();
 }
diff --git a/browser/ui/views/sidebar/sidebar_container_view.h b/browser/ui/views/sidebar/sidebar_container_view.h
index f0a2180915..7e36fd4f9b 100644
--- a/browser/ui/views/sidebar/sidebar_container_view.h
+++ b/browser/ui/views/sidebar/sidebar_container_view.h
@@ -98,6 +98,7 @@ class SidebarContainerView
   void ShowOptionsEventDetectWidget(bool show);
   void ShowSidebar(bool show_sidebar, bool show_event_detect_widget);
   SidebarShowOptionsEventDetectWidget* GetEventDetectWidget();
+  bool IsPanelActive() const;
   bool ShouldShowSidebar() const;
   void UpdateToolbarButtonVisibility();
 
```

Reply:
Hmm, I don't love the `SetSidebarShowOption` function having to know under which circumstance the bar should be hidden immediately (i.e. `IsPanelActive`) - seems better if that's handled more centrally. I'm looking to see if we can get an event when the context menu is closed, and call `ShowSidebar(ShouldShowSidebar())`